lib/dialects/mssql/async-queue.ts

```diff
@@ -0,0 +1,59 @@
+import { BaseError, ConnectionError } from '../../errors';
+
+/**
+ * Thrown when a connection to a database is closed while an operation is in progress
+ */
+export class AsyncQueueError extends BaseError {
+  constructor(message: string) {
+    super(message);
+    this.name = 'SequelizeAsyncQueueError';
+  }
+}
+
+class AsyncQueue {
+  previous: Promise<unknown>;
+  closed: boolean;
+  rejectCurrent: (reason?: any) => void;
```

Comment:
bit of a nit: can you specify `private` / `protected` as needed here (and everywhere else)?

Reply:
I personally also prefer explicitly setting `public` but I don't feel super strongly on that.

Reply:
I would keep these properties as is, since they weren't prefixed with an underscore. I think most properties in `lib/errors`, if not all, are meant to be public